### PR TITLE
Make dns daemonset, configmap and service creation specific to given custom resource

### PR DESCRIPTION
**Pull request by pravisankar**
- This will help us in handling deletion events. We will know exact resources to delete.
- This also helps to support multiple dns deployments with different configs if needed.

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge PTAL

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -130,18 +131,36 @@ func (f *Factory) DNSConfigMap(dns *dnsv1alpha1.ClusterDNS) (*corev1.ConfigMap,
...
+
+	if ds.Spec.Selector.MatchLabels == nil {
+		ds.Spec.Selector.MatchLabels = map[string]string{}
+	}
+	ds.Spec.Selector.MatchLabels["dns"] = ds.Name
+
+	if len(ds.Spec.Template.Spec.Volumes) == 0 {
+		return nil, fmt.Errorf("expected volume for core file not found")
+	}
+	ds.Spec.Template.Spec.Volumes[0].ConfigMap.Name = ds.Name
```
Comment (ironcladlou) [inline pkg/manifests/manifests.go]: Look it up by name (config-volume) instead of index?

**Comment by Miciah** [comment]
Might as well update `assets/dns/service.yaml` to add a `# Name is set at runtime` comment as you have done for `assets/dns/configmap.yaml` and `assets/dns/daemonset.yaml`.

**Comment by pravisankar** [comment]
@ironcladlou @Miciah  updated, PTAL

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -130,18 +131,45 @@ func (f *Factory) DNSConfigMap(dns *dnsv1alpha1.ClusterDNS) (*corev1.ConfigMap,
...
+	ds.Spec.Template.Labels["dns"] = ds.Name
+
+	if ds.Spec.Selector.MatchLabels == nil {
+		ds.Spec.Selector.MatchLabels = map[string]string{}
+	}
+	ds.Spec.Selector.MatchLabels["dns"] = ds.Name
+
+	if len(ds.Spec.Template.Spec.Volumes) == 0 {
+		return nil, fmt.Errorf("no volumes found, expected volume for core file")
+	}
```
Comment (Miciah) [inline pkg/manifests/manifests.go]: This check is superfluous now.

**Comment by ironcladlou** [comment]
Thanks!

/lgtm

**Pull request merged**